daemon/inertia/auth/certificate.go

```diff
@@ -58,7 +57,8 @@ func pemBlockForKey(priv interface{}) (*pem.Block, error) {
 	}
 }
 
-func generateCertificate(certPath string, keyPath string, host string) error {
+// GenerateCertificate creates an SSL certificate for HTTPS use
+func GenerateCertificate(certPath, keyPath, host, ecdsaCurve string) error {
```

Comment:
Changes not really needed, but I have a weird feeling about the empty string as one of the switch cases. It maps directly to an RSA key when the function params is called `ecdsa`. Moreover, the message specified in the default case gives the false impression that `""` would map to an ECDSA curve. Substituting `"RSA"` for `""` and changing the error message could be an option. It would also help express the programmer's intent when they use it in the code, e.g. `err := GenerateCertificate(dir+"test.cert", dir+"test.key", "0.0.0.0", "RSA")`

What do you think?

Reply:
That's actually a really good point, didn't think about the error reporting nothing!

I will update this - thanks for pointing it out!

Reply:
Also consider modifying the error message? It currently implies that a valid ECDSA scheme is the only way to go.

Reply:
😅